src/boot/efi/stub.c

```diff
@@ -958,8 +992,21 @@ static void load_all_addons(
         if (err != EFI_SUCCESS)
                 log_error_status(err, "Error loading global addons, ignoring: %m");
 
+        err = load_addons(
+                        image,
+                        loaded_image,
+                        u"\\loader\\vendor-addons", // TODO: I think there's a bug here. Bootspec looks in both ESP and XBOOTLDR for global addons/creds/etc. Meanwhile, the stub only looks at the device the UKI is on.
```

Comment:
can you file an issue about this. we indeed should probably look in both places. but it's not obvious how this should really work. i.e. if people directly boot into a UKI (i.e. without sd-boot) I think we probably shouldn't look into any partition, but if people go via sd-boot we probably should. hence maybe sd-boot should communicate this via an efi var somehow, so that the stub just checks that efi var for a secondary place.